gtsfm/two_view_estimator.py

```diff
@@ -85,19 +90,74 @@ def create_computation_graph(
 
         # if we have the expected data, evaluate the computed relative pose
         if i2Ti1_expected_graph is not None:
-            error_graphs = dask.delayed(compute_relative_pose_metrics)(i2Ri1_graph, i2Ui1_graph, i2Ti1_expected_graph)
+            pose_error_graphs = dask.delayed(compute_relative_pose_metrics)(
+                i2Ri1_graph, i2Ui1_graph, i2Ti1_expected_graph
+            )
+            corr_error_graph = dask.delayed(compute_correspondence_metrics)(
+                keypoints_i1_graph,
+                keypoints_i2_graph,
+                v_corr_idxs_graph,
+                camera_intrinsics_i1_graph,
+                camera_intrinsics_i2_graph,
+                i2Ti1_expected_graph,
+                self._corr_metric_dist_threshold,
+            )
         else:
-            error_graphs = (None, None)
+            pose_error_graphs = (None, None)
+            corr_error_graph = None
 
         return (
             i2Ri1_graph,
             i2Ui1_graph,
             v_corr_idxs_graph,
-            error_graphs[0],
-            error_graphs[1],
+            pose_error_graphs[0],
+            pose_error_graphs[1],
+            corr_error_graph,
         )
 
 
+def compute_correspondence_metrics(
+    keypoints_i1: Keypoints,
+    keypoints_i2: Keypoints,
+    corr_idxs_i1i2: np.ndarray,
+    intrinsics_i1: Cal3Bundler,
+    intrinsics_i2: Cal3Bundler,
+    i2Ti1: Pose3,
+    epipolar_distance_threshold: float,
+) -> Tuple[int, float]:
+    """Compute the metrics for the generated verified correspondence.
+
+    Args:
+        keypoints_i1: detected keypoints in image i1.
+        keypoints_i2: detected keypoints in image i2.
+        corr_idxs_i1i2: indices of correspondences.
+        intrinsics_i1: intrinsics for i1.
+        intrinsics_i2: intrinsics for i2.
+        i2Ti1: relative pose.
+        epipolar_distance_threshold: max epipolar distance to qualify as a correct match.
+
+    Returns:
+        Number of correct correspondences.
+        Inlier Ratio, i.e. ratio of correspondences which are correct.
+    """
+    number_correct = metric_utils.count_correct_correspondences(
+        keypoints_i1.extract_indices(corr_idxs_i1i2[:, 0]),
+        keypoints_i2.extract_indices(corr_idxs_i1i2[:, 1]),
+        intrinsics_i1,
+        intrinsics_i2,
+        i2Ti1,
+        epipolar_distance_threshold,
+    )
+
+    # logger.debug(
```

Comment:
did we mean to comment this one out?

Reply:
should be removed, as we are not logging them anymore.